sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/SecondActivity.kt

```diff
@@ -0,0 +1,112 @@
+package io.sentry.samples.android
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import io.sentry.Sentry
+import io.sentry.SpanStatus
+import io.sentry.samples.android.databinding.ActivitySecondBinding
+import okhttp3.OkHttpClient
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class SecondActivity : AppCompatActivity() {
+
+    private lateinit var repos: List<Repo>
+
+    private val client = OkHttpClient.Builder().addInterceptor(NetworkInterceptor()).build()
+
+    private lateinit var binding: ActivitySecondBinding
+
+    private val retrofit = Retrofit.Builder()
+            .baseUrl("https://api.github.com/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(client)
+            .build()
+
+    private val service = retrofit.create(GitHubService::class.java)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val activeSpan = Sentry.getSpan()
+        val span = activeSpan?.startChild("onCreate", javaClass.simpleName)
+
+        binding = ActivitySecondBinding.inflate(layoutInflater)
+
+        binding.doRequest.setOnClickListener {
+            updateRepos()
+        }
+
+        binding.backMain.setOnClickListener {
+            finish()
+            startActivity(Intent(this, MainActivity::class.java))
+        }
+
+        // do some stuff
+
+        setContentView(binding.root)
+
+        span?.finish(SpanStatus.OK)
+    }
+
+    private fun showText(visible: Boolean = true, text: String = "") {
+        binding.text.text = if (visible) text else ""
+        binding.text.visibility = if (visible) View.VISIBLE else View.GONE
+
+        binding.indeterminateBar.visibility = if (visible) View.GONE else View.VISIBLE
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        val span = Sentry.getSpan()?.startChild("onResume", javaClass.simpleName)
```

Comment:
`m`: Isn't this something we should do, and you shouldn't have to do manually?

Reply:
we already discussed this in a call, ideally, yes, not now, this is a sample activity